pandas/plotting/_core.py

```diff
@@ -815,11 +816,22 @@ def _post_plot_logic(self, ax, data):
 class ScatterPlot(PlanePlot):
     _kind = 'scatter'
 
-    def __init__(self, data, x, y, s=None, c=None, **kwargs):
+    def __init__(self, data, x, y, s=None, s_grow=1, c=None, **kwargs):
```

Comment:
I'm wary of adding this in the middle of the signature.  If people pass in arguments by position, this will cause their code to break because now `s_grow` will be assigned the value instead of `c`.

Reply:
Should I make s_grow the last positionnal argument? Or make it a keyword argument?

Reply:
Keyword, just after `c=None`.

Reply:
I meant to ask if you wanted s_grow to be the last keyword argument, like so:

``` python
def __init__(self, data, x, y, s=None, c=None, s_grow=1, **kwargs):
    # [...]
```

or if s_grow shoulb be a keyword-only argument, included in the **kwargs:
``` python
def __init__(self, data, x, y, s=None, c=None, **kwargs):
    if 's_grow' in kwargs:
        #[...]
```

I understand you're OK with the first one?

Reply:
Correct, I would have to look more closely, but `**kwargs` is usually passed through to the underlying matplotlib function.

The function signatures in `.plot` and `.plot.scatter` are a little messy, so you'll need to be careful to handle `**kwargs` appropriately. Having `s_grow=1` in the function signature is a good way to ensure that it isn't accidentally passed through to matplotlib.

Reply:
the name seems awkward

does seaborn/mpl have a comment name for this?

Reply:
I'm not very happy with the name s_grow either... Maybe size_factor is better?

I don't find anything similar in seaborn or matplotlib.

In ggplot2 there does not seem to be an option to pass a **scaling factor** to bubble plots, but you have the option to pass a size **range** instead. The argument name is scale_size (which I don't find to be a very good name either!).

http://t-redactyl.io/blog/2016/02/creating-plots-in-r-using-ggplot2-part-6-weighted-scatterplots.html

Having the option to specify a size range is more flexible than only being able to specify a scaling factor, as it allows to visualize small variations of data which would be invisible in a bubble plot with bubble areas proportional to the data, but the down side is precisely that it breaks the proportionality between data and bubble areas, which can result in unintentionnaly misleading and untrustworthy visualizations where data points with small relative differences are represented by very different bubble sizes.

As a Pandas user I would prefer having the option to pass a **scaling factor** rather than a **size range**.

What are your thoughts?
*  Scaling factor (like what I implemented) or size range (like in ggplot2) ?
* If we keep things as they are (scaling factor), should I replace s_grow by size_factor? Other name?